xarray/core/dataset.py

```diff
@@ -210,6 +224,7 @@ def __init__(self, data_vars=None, coords=None, attrs=None,
             coords = {}
         if data_vars is not None or coords is not None:
             self._set_init_vars_and_dims(data_vars, coords, compat)
+            self._check_multiindex_level_names()
```

Comment:
Rather than putting this check in the `DataArray` and `Dataset` constructors, let's calling this from the `Variable` constructor instead. Probably the cleanest place to put this check in `PandasIndexAdapter`, which is already used to wrap the data from all pandas Indexes:
https://github.com/pydata/xarray/blob/master/xarray/core/indexing.py#L434